### ISSUE TITLE
test case missing for pointless-statement

### ISSUE THREAD

**Issue by sushobhit27** [PyCQA/pylint#2213]
### Steps to reproduce
1. $ cat sample.py
```python
seq = list(range(10))
a = seq[5: 2: ]
```
2. $ pylint sample.py 

### Current behavior
-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 3.33/10, +6.67)

### Expected behavior
line 2 is assigning empty list to b, therefore it should give:
sample.py:2:0: W0104: Statement seems to have no effect (pointless-statement)

or it can be a new check also like "slicing-return-nothing".
slicing never give error as if it can't slice anything from input, it just returns empty list.
In general whenever slicing returns empty list, this error should be raised.
All three values start, stop and step will be required for this to work. e.g
```python
>>> a = list(range(10))
>>> a[5: 2]
[]
>>> a[-5: -2]
[5, 6, 7]
>>> a[-2: -5]
[]
>>> a[-2: -5: -1]
[8, 7, 6]
```
### pylint --version output
pylint 2.0.0.dev1
astroid 2.0.0.dev1
Python 3.6.4 (default, Jan  7 2018, 15:53:53)

**Comment by PCManticore** [comment]
This sounds interesting, but can be pretty hard to infer correctly most of the them outside of artificial examples, since we'll have to infer the actual list to get its bounds. For me it feels like a potential source of false positives, especially since in some cases we infer an empty list because we don't have better information available.

**Comment by sushobhit27** [comment]
@PCManticore No need to infer the actual list, we just needed start, stop and step.
Empty list is returned, whenever start > stop if step is +ve or start < stop if step is -ve.
The main problem could have been, if the slicing involved variables, e.g
start = 5
stop = 2
seq[start: stop]
In this case, how to infer the correct value of start & stop, as these are now astroid.Name instead of astroid.Const. I tried to figure out getting value from astroid.Name but couldn't.

**Comment by PCManticore** [comment]
There's also to consider if this is feasible for numpy arrays, where the slicing might behave differently then that of a normal Python list. The surface area of false positive is too big for little benefit IMO.

**Comment by sushobhit27** [comment]
Hmm, thats true and better not to have new functionality if it start giving false positive in existing one.
